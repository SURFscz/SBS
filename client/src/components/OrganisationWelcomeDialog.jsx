import React, {useState} from "react";
import I18n from "../locale/I18n";
import "./WelcomeDialog.scss";
import OrganisationEn from "./welcome/OrganisationEn";
import OrganisationNl from "./welcome/OrganisationNl";
import {ROLES} from "../utils/UserRole";
import ToggleSwitch from "./redesign/ToggleSwitch";
import {AlertType, Modal} from "@surfnet/sds";

export default function OrganisationWelcomeDialog({
                                                      name,
                                                      role,
                                                      isAdmin = false,
                                                      isOpen = false,
                                                      close,
                                                      isInvitation
                                                  }) {

    const [toggleRole, setToggleRole] = useState(role);

    const doToggleRole = () => {
        setToggleRole(toggleRole === ROLES.ORG_ADMIN ? ROLES.ORG_MANAGER : ROLES.ORG_ADMIN);
    }

    const content = () => {
        return (
            <section className={"welcome-dialog-content"}>
                {(isAdmin && !isInvitation) && <div className="toggle-role">
                    <ToggleSwitch value={toggleRole === ROLES.ORG_ADMIN} onChange={doToggleRole}/>
                    <span className="toggle-txt">{I18n.t("welcomeDialog.toggleRole")}</span>
                </div>}
                <section className="responsibilities">
                    {I18n.locale === "en" ? <OrganisationEn role={toggleRole}/> : <OrganisationNl role={toggleRole}/>}
                </section>
            </section>
        );
    }
    if (!isOpen) {
        return null;
    }

    return (
        <Modal
            confirm={close}
            alertType={AlertType.Info}
            subTitle={I18n.t(`welcomeDialog.${toggleRole === ROLES.ORG_ADMIN ? "roleOrganisationAdmin" : "roleOrganisationManager"}`)}
            children={content()}
            title={I18n.t("welcomeDialog.title", {name: name})}
            confirmationButtonLabel={I18n.t("welcomeDialog.ok", {type: I18n.t("welcomeDialog.organisation")})}
            className={"welcome-dialog"}
        />
    );
}

