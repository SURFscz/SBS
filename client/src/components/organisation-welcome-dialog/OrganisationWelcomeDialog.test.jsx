import React from "react";
import {render} from "@testing-library/react";
import OrganisationWelcomeDialog from "./OrganisationWelcomeDialog";

describe("OrganisationWelcomeDialog", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(
            <OrganisationWelcomeDialog
                open={true}
                onClose={() => {}}
                onCreate={() => {}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom title", () => {
        const {asFragment} = render(
            <OrganisationWelcomeDialog
                open={true}
                onClose={() => {}}
                onCreate={() => {}}
                title="Custom Title"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom description", () => {
        const {asFragment} = render(
            <OrganisationWelcomeDialog
                open={true}
                onClose={() => {}}
                onCreate={() => {}}
                description="Custom Description"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
