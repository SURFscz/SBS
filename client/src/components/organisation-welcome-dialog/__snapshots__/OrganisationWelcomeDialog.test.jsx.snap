// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationWelcomeDialog matches snapshot with custom description 1`] = `<DocumentFragment />`;

exports[`OrganisationWelcomeDialog matches snapshot with custom title 1`] = `<DocumentFragment />`;

exports[`OrganisationWelcomeDialog matches snapshot with default props 1`] = `<DocumentFragment />`;
